wolvic/browser/vr/wvr_manager.cc

```diff
@@ -157,16 +157,29 @@ device::mojom::XRViewPtr CreateView(
   return view;
 }
 
+gfx::Size GetSuggestedFrameSize(WvrApi* wvr_api) {
+  // Make sure we're fetching an up to date state.
+  wvr_api->PullSystemState();
+
+  const mozilla::gfx::VRDisplayState& ds =
+      wvr_api->get_system_state().displayState;
+
+  // We compute the suggested frame size based on eye resolution.
+  gfx::Size frame_size(ds.eyeResolution.width * 2, ds.eyeResolution.height);
+  return frame_size;
+}
+
 std::vector<device::mojom::XRViewPtr> CreateViews(
-    const mozilla::gfx::VRDisplayState& display_state,
-    const mozilla::gfx::VRPose* pose,
-    gfx::Size maximum_size) {
+    WvrApi* wvr_api, const mozilla::gfx::VRPose* pose) {
+  const mozilla::gfx::VRDisplayState& display_state =
+    wvr_api->get_system_state().displayState;
+  gfx::Size suggested_size = GetSuggestedFrameSize(wvr_api);
+
   std::vector<device::mojom::XRViewPtr> views(2);
   views[0] = CreateView(mozilla::gfx::VRDisplayState::Eye::Eye_Left,
-                        display_state, maximum_size, pose);
+                        display_state, suggested_size, pose);
```

Comment:
CreateView() argument is still called 'maximum_size', should we also rename it?

Reply:
I think it's OK because it's really a max size. We're passing here just a smaller (on average) value.